File: ProgramowanieObiektowe/Zadanie2/files.cs
using System;
using System.IO;
using System.Linq;

namespace ProgramowanieObiektowe2
{
    class MainClass
    {
        public static void Main(string[] args)
        {
            Console.WriteLine("Podaj nazwe pliku");
            string name = Console.ReadLine();
            StreamReader sr = new StreamReader(name + ".txt");
            //try if file was opened correctly
            string[] lines = sr.ReadToEnd().Split('\n');
            float max_accuracy=-1;
            //check if line is empty
            foreach (var line in lines)
            {
                Console.WriteLine(line);
                string[] words = line.Split(' ');
                float accuracy = Convert.ToSingle(words[3]); 
                if(accuracy>max_accuracy)
                {
                    max_accuracy = accuracy;
                }
            }
            StreamWriter sw = new StreamWriter(name + "max.txt");
            foreach (var line in lines)
            {
                string[] words = line.Split(' ');
                float accuracy = Convert.ToSingle(words[3]); 
                if(accuracy.Equals(max_accuracy) && words[0].Length>=3 && words[1].EndsWith("ski", StringComparison.Ordinal))
                {
                    //write name
                    sw.Write(words[0][0]);
                    for (int i = 0; i < words[0].Length - 4; i++)
                        sw.Write('*');
                    if(words[0].Length==3)
                        sw.Write(words[0].Substring(words[0].Length - 2) + ' ');
                    else sw.Write(words[0].Substring(words[0].Length-3)+' ');
                    //write surname
                    sw.Write(words[1][0]);
                    for (int i = 0; i < words[1].Length - 4; i++)
                        sw.Write('*');
                    sw.Write(words[1].Substring(words[1].Length - 3)+' ');
                    sw.Write(words[2]);
                }
            }
            sw.Close();

        }
    }
}

File: ProgramowanieObiektowe/Zadanie2/files.py
name = input("Podaj nazwe pliku ")
file = open(name+".txt")
file_to_save = open(name+"max.txt", 'w')
accuracy = []
try:
    for line in file:
        words = line.split(' ')
        accuracy.append(words[3])
    file.seek(0)
    for line in file:
        words = line.split(' ')
        if(float(words[3])==float((max(accuracy))) and len(words[0])>=3 and words[1][-3:]=="ski"):
            a=[]
            a.append(words[0][0])
            for i in range(len(words[0])-4):
                a.append('*')
            if(len(words[0])==3):
                a.append(words[0][-2:])
            else:
                a.append(words[0][-3:])
            answer=''.join(a)
            file_to_save.write(answer+' ')
            a.clear()
            a.append(words[1][0])
            for i in range(len(words[1]) - 4):
                a.append('*')
            if (len(words[1]) == 3):
                a.append(words[1][-2:])
            else:
                a.append(words[1][-3:])
            answer = ''.join(a)
            file_to_save.write(answer)
            file_to_save.write(' '+words[2])

except:
    print("blad pliku")
